README.adoc

```diff
@@ -88,85 +88,122 @@ The `withGradle` wrapper should be used instead, since it also deals well with p
+
+The same auto-injection behavior is available for the https://github.com/gradle/common-custom-user-data-gradle-plugin[Common Custom User Data Gradle plugin] and the https://github.com/gradle/common-custom-user-data-maven-extension[Common Custom User Data Maven extension].
 
 > **Note**
 > - The configuration is based on global environment variables and applies to all builds on all connected agents matching the specified label criteria, or all in case no label criteria are defined.
 
-=== How to enable Build Scan injection
+=== Enable auto-injection
+
+The auto-injection is split into three levels to have fine-grained control over the enablement of this feature.
 
-- Set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION` environment variable to any value.
-- Set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_URL` environment variable to the URL of the Gradle Enterprise instance to which to publish the Build Scan.
+==== Global
+
+To globally enable the auto-injection set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_INJECTION` environment variable to `true`.
+Next set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_URL` environment variable to the URL of the Gradle Enterprise instance.
 
 ==== Gradle
 
-- Set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION` environment variable to the desired Gradle Enterprise plugin version to be injected.
-- Optionally set the `JENKINSGRADLEPLUGIN_CCUD_PLUGIN_VERSION` environment variable to the desired version of the Common Custom User Data Gradle plugin to be injected.
+To enable the auto-injection for Gradle builds, set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_PLUGIN_VERSION` environment variable to the desired https://docs.gradle.com/enterprise/gradle-plugin/[Gradle Enterprise Gradle plugin] version to be used.
+
+Optionally set the `JENKINSGRADLEPLUGIN_CCUD_PLUGIN_VERSION` environment variable to the desired version of the https://github.com/gradle/common-custom-user-data-gradle-plugin[Common Custom User Data Gradle plugin] to be used.
 
 ==== Maven
 
-- Set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION` environment variable to any value to enable the Gradle Enterprise Maven extension injection.
-Currently, this environment variable value is ignored and is used just as a trigger to enable injection. To see which version of the extension is injected, see link:#_compatibility[Compatibility].
-- Optionally set the `JENKINSGRADLEPLUGIN_CCUD_EXTENSION_VERSION` to any value to enable Common Custom User Data Maven extension injection.
-Currently, this environment variable value is ignored and is used just as a trigger to enable injection. To see which version of the extension is injected, see link:#_compatibility[Compatibility].
+To enable the auto-injection for Maven builds, set the `JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_EXTENSION_VERSION` environment variable to any value to enable the https://docs.gradle.com/enterprise/maven-extension/[Gradle Enterprise Maven extension] injection.
```

Comment:
=> To enable the auto-injection .... to enable the injection
This sentence looks weird